File: HomeWork05/Menu.java

import java.util.ArrayList;

public class Menu {
    private ArrayList<PunktMenu> puncts;
    public int countPunkts;

    public Menu() {
        this.puncts = new ArrayList<>();
        this.countPunkts = 0;
    }

    public void addPunkt(PunktMenu punktMenu) {
        if (puncts.add(punktMenu))
            countPunkts++;
    }

    public void clear() {
        this.puncts = new ArrayList<>();
        this.countPunkts = 0;
    }

    public PunktMenu getPunkt(int number){
        if (number >0 && number <= countPunkts){
            return this.puncts.get(number-1);
        } else {
            return null;
        }
        
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        int i = 1;
        sb.append("Меню:\n");
        for (PunktMenu punktMenu : puncts) {
            sb.append(String.format("%d. %s\n", i++, punktMenu));
        }
        return sb.toString();
    }
}

File: HomeWork07/Models/Menu/Menu.cs
using System.Text;

namespace HomeWork07;

public class Menu {
    private List<PunktMenu> puncts;

    public Menu() {
        this.puncts = new List<PunktMenu>();
    }

    public void addPunkt(PunktMenu punktMenu) {
        puncts.Add(punktMenu);
    }

    public void clear() {
        this.puncts = new List<PunktMenu>();
        
    }

    public PunktMenu getPunkt(int number){
        foreach (PunktMenu item in puncts)
        {   
            if (item.number == number) {
                return item;
            }        
        }
        return null;
    }

    public override string ToString() {
        StringBuilder sb = new StringBuilder();
        if (puncts.Count>0)
        sb.Append("Меню:\n");
        foreach (PunktMenu pm in puncts) {
            sb.Append(string.Format($"{pm.ToString()} \n"));
        }
        return sb.ToString();
    }
}
